### PR TITLE
Added passing user data from Register and Edit forms to db

### PR DESCRIPTION
**Pull request by MikolekN**
Added get_client_by_mail method to be able to get user data based on the user's email.
According to the requirements: "Klient może być identyfikowany np. poprzez e-mail" and "Po wpisaniu identyfikatora (adresu e-mail)" the user should be identified by their email.
Shouldn't the database also use email as an identifier then?

### CODE REVIEW

**Review by dlydka — APPROVED** [review]

File: src/csvDatabase.py [hunk 1]
```diff
@@ -75,6 +75,20 @@ def get_client(self, client_id: str) -> Dict | None:
 
         return None
 
+    def get_client_by_mail(self, client_email: str) -> Dict | None:
```
Comment (Lenetis) [inline src/csvDatabase.py]: I'm not sure if this method is necessary. The same thing could be achieved by using the get_clients() method and providing the email in the predicate. E.g.:
`client = database.get_clients(lambda client: client["mail"] == some_mail)[0]`

File: src/user_dto.py [hunk 2]
```diff
@@ -19,44 +14,58 @@ class UserDto:
...
     surname: str
+    age: int
     email: str
-    gender: Gender
+    gender: str
+    id: int = None
+
+    def __post_init__(self):
+        if self.id is None:
+            self.id = random.randint(1, 1000)
```
Comment (Lenetis) [inline src/user_dto.py]: This has a small chance of collisions. I'd suggest using the uuid module:
```python
import uuid

self.id = uuid.uuid4().int  # 128-bit integer
# or alternatively
self.id = uuid.uuid4().hex  # hex string
```

**Review by Lenetis — APPROVED** [review]
Approved, but please consider the comments.
Also, an idea - since mails are supposed do be unique, couldn't they be used as the database ids?

File: src/csvDatabase.py (hunk 1 above)
Comment (MikolekN) [reply src/csvDatabase.py]: That's true. I've overlooked this method while reviewing the code and have made the required adjustments in my code.

**Pull request merged**